README.adoc

```diff
@@ -1101,6 +1101,23 @@ User.where.not(status: 'active', plan: 'basic')
 User.where.not('status = ? AND plan = ?', 'active', 'basic')
 ----
 
+=== Redundant `all` [[redundant-all]]
+Using `all` as a receiver for Active Record query methods is redundant. 
+The result won't change without `all`, so it can be removed.
+
+[source, ruby]
+----
+# bad
+User.all.order(:created_at)
+User.all.find(id)
```

Comment:
Is it only applicable to the model or to a relation as well? ‘current_user.likes.all’?

Reply:
Yes, this applies to relation objects as well.

```ruby
# bad
current_user.likes.all
current_user.all.order(:created_at)
current_user.likes.all.order(:created_at)

# good
current_user.likes
current_user.order(:created_at)
current_user.likes.order(:created_at)
```

Reply:
True, examples of `all` method followed by an AR query method could be added.

Reply:
@koic
> True, examples of all method followed by an AR query method could be added.

Sorry for the confusion, but I'd like to clarify something regarding your comment.
Are you suggesting that I should add one or both of the following examples?

(`like` is a relation of User)
- 1. `relation.all.ar_query_method`
  ```ruby
  # bad
  current_user.likes.all.order(:created_at)
  
  # good
  current_user.likes.order(:created_at)
  ```

- 2. `relation.ar_query_method.all`

  ```ruby
  # bad
  current_user.likes.order(:created_at).all
  
  # good
  current_user.likes.order(:created_at)
  ```

Reply:
I imagined it as 1. Either way, the receiver is an `ActiveRecord_Relation` object.

Reply:
Thank you for your comment. I've added some examples.